src/traits.rs

```diff
@@ -21,8 +21,28 @@ impl<N> NodeIdT for N where N: Eq + Ord + Clone + Debug + Hash + Send + Sync {}
 pub trait Message: Debug + Send + Sync {}
 impl<M> Message for M where M: Debug + Send + Sync {}
 
-/// Result of one step of the local state machine of a distributed algorithm. Such a result should
-/// be used and never discarded by the client of the algorithm.
+/// Single algorithm step result.
+///
+/// Each time input (typically in the form of user input or incoming network messages) is provided
+/// to an instance of an algorithm, a `Step` is produced as a result, potentially containing
+/// output values, fault log and network messages.
+///
+/// Any `Step` **must always be used** by the client application; at the very least the resulting
+/// messages must be queued.
+///
+/// ## Handling unused Steps
+///
+/// In the (rare) case of a `Step` not being of any interest at all, instead of discarding it
+/// through `let _ = ...` or similar constructs, the implicit assumption should explicitliy be
+/// checked instead:
+///
+/// ```rust,norun
+/// assert!(alg.propose(123).expect("Could not propose value").is_empty(),
+///         "Algorithm will never output anything on first proposal");
```

Comment:
I know it's just for the example, but: If we are the only validator, the first proposal usually _will_ cause an immediate output.

Reply:
I know, I found out the hard way! `alg` is neutral though, doesn't have to be HoneyBadger =).

I can't think of an example where it would be valid to not handle the step at the moment.